packages/password/src/index.mts

```diff
@@ -1,39 +1,98 @@
-import type { Prompt } from '@inquirer/type';
-import input from '@inquirer/input';
+import {
+  createPrompt,
+  useState,
+  useKeypress,
+  usePrefix,
+  isEnterKey,
+  type PromptConfig,
+} from '@inquirer/core';
 import chalk from 'chalk';
 import ansiEscapes from 'ansi-escapes';
 
-type InputConfig = Parameters<typeof input>[0];
-type PasswordConfig = Omit<InputConfig, 'transformer' | 'default'> & {
+type PasswordConfig = PromptConfig<{
   mask?: boolean | string;
-};
+  validate?: (value: string) => boolean | string | Promise<string | boolean>;
+  allowShowPassword?: boolean;
+}>;
 
-const password: Prompt<string, PasswordConfig> = (config, context) => {
+export default createPrompt<string, PasswordConfig>((config, done) => {
   if ('transformer' in config) {
     throw new Error(
       'Inquirer password prompt do not support custom transformer function. Use the input prompt instead.',
     );
   }
 
-  return input(
-    {
-      ...config, // Make sure we do not display the default password
-      default: undefined,
-      transformer(str: string, { isFinal }: { isFinal: boolean }) {
-        if (config.mask) {
-          const maskChar = typeof config.mask === 'string' ? config.mask : '*';
-          return maskChar.repeat(str.length);
-        }
-
-        if (!isFinal) {
-          return `${chalk.dim('[input is masked]')}${ansiEscapes.cursorHide}`;
-        }
-
-        return '';
-      },
-    },
-    context,
-  );
-};
-
-export default password;
+  const { validate = () => true, allowShowPassword } = config;
+  const [status, setStatus] = useState<string>('pending');
+  const [errorMsg, setError] = useState<string | undefined>(undefined);
+  const [value, setValue] = useState<string>('');
+  const [isMasked, setIsMasked] = useState<boolean>(true);
+
+  const isLoading = status === 'loading';
+  const prefix = usePrefix(isLoading);
+
+  useKeypress(async (key, rl) => {
+    // Ignore keypress while our prompt is doing other processing.
+    if (status !== 'pending') {
+      return;
+    }
+
+    if (isEnterKey(key)) {
+      const answer = value;
+      setStatus('loading');
+      const isValid = await validate(answer);
+      if (isValid === true) {
+        setValue(answer);
+        setStatus('done');
+        done(answer);
+      } else {
+        // Reset the readline line value to the previous value. On line event, the value
+        // get cleared, forcing the user to re-enter the value instead of fixing it.
+        rl.write(value);
+        setError(isValid || 'You must provide a valid value');
+        setStatus('pending');
+      }
+    } else if (allowShowPassword && key.ctrl === true && key.name === '`') {
+      // CTRL + Space
+      // I only tried on Linux, but the combination on Linux was reported like that
+      //   key.crtl = true
+      //   key.name = '`'
```

Comment:
Sooo, did some more digging into this. And it turned out to be a bit of a headache.

Depending on the environment and the terminal used, there can be a lot of conflict.

For one, `CTRL + Space` for me open the AI helper sidebar and never reach the node program.

I then started to think of keys; like `CTRL+H`, but that's a backspace shortcut (and it doesn't even register as the H key.)

So long story short:

1. I added `node tools/keys.mjs` to the repo latest version to play with.
2. Potential solution 1: finding a key combo that'll reasonably work for most environment (terminals, OS & consider non-english keyboards).
3. Have a basic OS switch and use keys per OS that are reasonably going to work.